relm4-components/src/open_button/factory.rs

```diff
@@ -11,53 +11,33 @@ pub(crate) struct FileListItem {
     pub(crate) path: PathBuf,
 }
 
-#[derive(Debug)]
-pub(crate) struct FileListItemWidgets {
-    label: gtk::Button,
-    row: gtk::ListBoxRow,
-}
-
-impl FactoryPrototype for FileListItem {
-    type Factory = FactoryVecDeque<Self>;
-    type View = gtk::Box;
+#[relm4::factory(pub(crate))]
+impl FactoryComponent for FileListItem {
+    type ParentInput = OpenButtonMsg;
+    type CommandOutput = ();
+    type Input = ();
+    type Init = PathBuf;
+    type ParentWidget = gtk::Box;
+    type Output = OpenButtonMsg;
     type Widgets = FileListItemWidgets;
-    type Root = gtk::ListBoxRow;
-    type Msg = OpenButtonMsg;
 
-    fn init_view(&self, key: &DynamicIndex, sender: relm4::Sender<Self::Msg>) -> Self::Widgets {
-        let label = gtk::Button::with_label(
-            self.path
-                .iter()
-                .last()
-                .expect("Empty path")
-                .to_str()
-                .expect("Couldn't convert path to string"),
-        );
-        let row = gtk::ListBoxRow::builder().child(&label).build();
-
-        label.inline_css(b"margin: 0");
-
-        let key = key.clone();
-        label.connect_clicked(move |_| {
-            send!(sender, OpenButtonMsg::OpenRecent(key.clone()));
-        });
-        FileListItemWidgets { label, row }
+    view! {
+        gtk::ListBoxRow {
+            gtk::Button {
+                set_label: self.path.iter().last().expect("Empty path").to_str().unwrap(),
+                inline_css: "margin: 0;",
```

Comment:
This should rather use `set_margin_all` from WidgetPlus.

Reply:
Fixed.